### PR TITLE
Add Lock/Unlock App with biometric or screenlock credentials

### PR DESCRIPTION
**Pull request by neopilou**
Add Lock/Unlock App with biometric or screenlock credentials (for non biometric capable device) : 

Option in App Settings :
Screenshot_20200228-141217_Home Assistant

When open app with active unlock setting :
Screenshot_20200228-134518_Home Assistant

For those who have a device allowing several types of biometric recognition, the preferred biometric (in the device settings) is used (but for those who have disfigured or cut their finger ;-} , the PIN code of the device can be used to unlock the app):
Screenshot_20200228-140841_Home Assistant

Screenshot_20200228-140949_Home Assistant

### CODE REVIEW

**Comment by dshokouhi** [comment]
This will close #112  :)

**Comment by JBassett** [comment]
After looking into the documentation more do we need to write our own pin entry stuff?  I feel like we are adding a lot of code when really we should just be using the device authentication.

https://developer.android.com/training/sign-in/biometric-auth#biometric-or-lock-screen

**Comment by neopilou** [comment]
Indeed, the entire "PIN code" part of the app can be removed. I just tested on an emulated device and without a biometric sensor, the screenlock  credential is asked (pattern, pin or  password). 
i'll do delete it.

File: app/src/main/java/io/homeassistant/companion/android/lock/LockActivity.kt [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                    override fun onAuthenticationSucceeded(
+                        result: BiometricPrompt.AuthenticationResult
+                    ) {
+                        super.onAuthenticationSucceeded(result)
+                        if (!set)
+                            lockViewPresenter?.onViewReady()
+                    }
+                })
+            val promptInfo = BiometricPrompt.PromptInfo.Builder()
+                .setTitle(fragment.resources.getString(R.string.biometric_title))
```
[the remark this answers is not in the document]
Comment (neopilou) [reply app/src/main/java/io/homeassistant/companion/android/lock/LockActivity.kt]: I can't, because it causes a type mismatch error.

**Comment by ncodee** [comment]
It's great that this has been implemented, but can we get something more neutral, the splash screen is quite obnoxious.

Also, I don't understand why it requires a 2nd tap to initiate the biometrics. It should be prompted on when first opened.

**Comment by dshokouhi** [comment]
@ncodee please don't comment on merged PR's, feel free to open an enhancement request for your desired functionality.

**Pull request merged**